src/engine/strat_engine/blockdev.rs

```diff
@@ -172,30 +169,29 @@ pub fn initialize(pool_uuid: &PoolUuid,
     fn filter_devs<I>(dev_infos: I,
                       pool_uuid: &PoolUuid,
                       force: bool)
-                      -> EngineResult<Vec<(Device, (PathBuf, u64))>>
-        where I: Iterator<Item = (Device, EngineResult<(PathBuf, u64, DevOwnership)>)>
+                      -> EngineResult<Vec<(Device, (PathBuf, u64, File))>>
+        where I: Iterator<Item = (Device, EngineResult<(PathBuf, u64, DevOwnership, File)>)>
     {
         let mut add_devs = Vec::new();
         for (dev, dev_result) in dev_infos {
             if dev_result.is_err() {
                 return Err(dev_result.unwrap_err());
             }
-            let (devnode, dev_size, ownership) = dev_result.unwrap();
+            let (devnode, dev_size, ownership, f) = dev_result.unwrap();
             if dev_size < MIN_DEV_SIZE {
                 let error_message = format!("{} too small, minimum {} bytes",
                                             devnode.display(),
                                             MIN_DEV_SIZE);
                 return Err(EngineError::Engine(ErrorEnum::Invalid, error_message));
             };
             match ownership {
-                DevOwnership::Unowned => add_devs.push((dev, (devnode, dev_size))),
+                DevOwnership::Unowned => add_devs.push((dev, (devnode, dev_size, f))),
                 DevOwnership::Theirs => {
                     if !force {
-                        let error_str = format!("First 4K of {} not zeroed", devnode.display());
+                        let error_str = format!("First 8K of {} not zeroed", devnode.display());
```

Comment:
It would be nice if this 8K were dependent on some constant, and not just a string, but I don't know that it is that important.